src/database/DataTypes/StocktakeBatch.js

```diff
@@ -45,26 +57,33 @@ export class StocktakeBatch extends Realm.Object {
    * Finalising StocktakeBatch will adjust inventory appropriately and will add
    * new TransactionBatch in reducing or increasing Transaction for this Stocktake
    * @param  {Realm}  database   App wide local database
-   * @param  {object} user       The user that finalised this stocktake
    */
   finalise(database) {
-    const isAddition = this.countedTotalQuantity > this.snapshotTotalQuantity;
-    const inventoryAdjustment = isAddition ? this.stocktake.getAdditions(database)
-                                            : this.stocktake.getReductions(database);
-    // Adjust inventory
+    // Update the itemBatch details
     this.itemBatch.batch = this.batch;
-    this.itemBatch.numberOfPacks = this.countedNumberOfPacks;
     this.itemBatch.expiryDate = this.expiryDate;
-    // Create TransactionItem, TransactionBatch to store inventory adjustment in this Stocktake
-    const item = this.itemBatch.item;
-    const transactionItem = createRecord(database, 'TransactionItem', inventoryAdjustment, item);
-    const transactionBatch = createRecord(database, 'TransactionBatch',
-                                          transactionItem, this.itemBatch);
-    transactionBatch.numberOfPacks = Math.abs(this.snapshotTotalQuantity
-                                              - this.countedTotalQuantity);
 
+    // Make inventory adjustments if there is a difference to apply
+    if (this.difference !== 0) {
```

Comment:
Check for this now, that way if user just updates batch details it's still included in the finalisation (these remain as lines 63-64 above).